src/VisualStudio/Core/Def/ProjectSystem/BrokeredService/WorkspaceProject.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Remote.ProjectSystem;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem.BrokeredService
+{
+    internal sealed class WorkspaceProject : IWorkspaceProject
+    {
+        // For the sake of the in-proc implementation here, we're going to build this atop IWorkspaceProjectContext so semantics are preserved
+        // for a few edge cases. Once the project system has moved onto this directly, we can flatten the implementations out.
+        private readonly IWorkspaceProjectContext _project;
+
+        public WorkspaceProject(IWorkspaceProjectContext project)
+        {
+            _project = project;
+        }
+
+        public ValueTask DisposeAsync()
+        {
+            _project.Dispose();
+            return ValueTaskFactory.CompletedTask;
+        }
+
+        public async Task AddAdditionalFilesAsync(IReadOnlyList<string> additionalFilePaths, CancellationToken cancellationToken)
+        {
+            await using var batch = _project.CreateBatchScope().ConfigureAwait(false);
```

Comment:
it sucks.  but the correct pattern is:

```c#
var batch = _project.CreateBatchScope().ConfigureAwait(false)
await using var _ = await batch.ConfigureAwait(false);
```

:'(

This is needed so that DisposeAsync is also called with CA(false).

Reply:
oh wait.... maybe this is waht your pattern is already doing?  not clear from reading which CA this is. so if this is correct, then awesome :)